l1-contracts/src/core/libraries/validator-selection/ValidatorSelectionLib.sol

```diff
@@ -181,49 +198,75 @@ library ValidatorSelectionLib {
     //  as we "start" time $t$.
     uint32 ts = Timestamp.unwrap(_epoch.toTimestamp()).toUint32() - 1;
     uint256 validatorSetSize = _stakingStore.attesters.lengthAtTimestamp(ts);
-
-    if (validatorSetSize == 0) {
-      return new address[](0);
-    }
-
     uint256 targetCommitteeSize = store.targetCommitteeSize;
 
+    bool smallerCommittee = validatorSetSize <= targetCommitteeSize;
+    address[] memory committee =
+      new address[](smallerCommittee ? validatorSetSize : targetCommitteeSize);
+
     // If we have less validators than the target committee size, we just return the full set
-    if (validatorSetSize <= targetCommitteeSize) {
-      return _stakingStore.attesters.valuesAtTimestamp(ts);
+    if (smallerCommittee) {
+      committee = _stakingStore.attesters.valuesAtTimestamp(ts);
```

Comment:
Would it not still be cleaner to just return it directly here instead of putting it into `committee` then going a bit more and then returning it, we don't need to prepare the first empty `committee` in that case really.

Edit: Actually just ignore for now, as this is some of the things that will change a little with GSE as we are calling out.